src/Aspire.Hosting.Azure.CosmosDB/AzureCosmosDBDatabaseResource.cs

```diff
@@ -35,11 +35,25 @@ public class AzureCosmosDBDatabaseResource(string name, string databaseName, Azu
     /// <summary>
     /// Gets the connection string expression for the Azure Cosmos DB database.
     /// </summary>
-    public ReferenceExpression ConnectionStringExpression => Parent.ConnectionStringExpression;
+    public ReferenceExpression ConnectionStringExpression =>
+        Parent.IsEmulator || Parent.UseAccessKeyAuthentication
+            ? ReferenceExpression.Create($"{Parent.ConnectionStringExpression};Database={DatabaseName}")
+            : ReferenceExpression.Create($"AccountEndpoint={Parent.ConnectionStringExpression};Database={DatabaseName}");
```

Comment:
Can't `AccountEndpoint` be added in `AzureCosmosDBResource`? This would move the logic dealing with this to one place.

Reply:
I was leaning towards not massaging things into a connection string until we absolutely needed it. So I do it here in the Database resource and don't use the conditional in the Container resource.

Reply:
I think it would be better if we followed the same pattern as EventHubs:

https://github.com/dotnet/aspire/blob/08b424e1fa0b28aa225a67e2c6a7b5b959037f05/src/Aspire.Hosting.Azure.EventHubs/AzureEventHubsResource.cs#L50-L82

https://github.com/dotnet/aspire/blob/08b424e1fa0b28aa225a67e2c6a7b5b959037f05/src/Aspire.Hosting.Azure.EventHubs/AzureEventHubConsumerGroupResource.cs#L35

The child resources call this method on the parent, and the parent builds up the connection string.